LanguageExt.Core/DataTypes/UnitsOfMeasure/Mass.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+
+namespace LanguageExt.Core.DataTypes.UnitsOfMeasure
+{
+  public struct Mass : IComparable<Mass>, IEquatable<Mass>
+  {
+    readonly double Value;
+    internal Mass(double value) =>
+        Value = value;
+
+    public override string ToString() =>
+        Kilograms + " kg";
+
+    #region CompareTo
+
+    public int CompareTo(Mass other) => Kilograms.CompareTo(other.Kilograms);
+
+    #endregion
+
+    #region Equals
+
+    public bool Equals(Mass other) => Kilograms.Equals(other.Kilograms);
+
+    public bool Equals(Mass other, double epsilon) => Math.Abs(other.Kilograms - Kilograms) < epsilon;
+
+    public override bool Equals(object obj) =>
+        obj == null
+            ? false
+            : obj is Mass
+                ? Equals((Mass)obj)
+                : false;
+
+    #endregion
+
+    public override int GetHashCode() => Kilograms.GetHashCode();
+
+    #region Basic maths functions
+
+    public Mass Add(Mass rhs) =>
+        new Mass(Kilograms + rhs.Kilograms);
+
+    public Mass Subtract(Mass rhs) =>
+        new Mass(Kilograms - rhs.Kilograms);
+
+    public Mass Multiply(double rhs) =>
+        new Mass(Kilograms * rhs);
+
+    public Mass Divide(double rhs) =>
+        new Mass(Kilograms / rhs);
+
+    #endregion
+
+    #region Operators
+
+    public static Mass operator *(Mass lhs, double rhs) =>
+        lhs.Multiply(rhs);
+
+    public static Mass operator *(double lhs, Mass rhs) =>
+        rhs.Multiply(lhs);
+
+    public static Mass operator +(Mass lhs, Mass rhs) =>
+        lhs.Add(rhs);
+
+    public static Mass operator -(Mass lhs, Mass rhs) =>
+        lhs.Subtract(rhs);
+
+    public static Mass operator /(Mass lhs, double rhs) =>
+        lhs.Divide(rhs);
+
+    public static double operator /(Mass lhs, Mass rhs) =>
+        lhs.Kilograms / rhs.Kilograms;
+
+    public static bool operator ==(Mass lhs, Mass rhs) =>
+        lhs.Equals(rhs);
+
+    public static bool operator !=(Mass lhs, Mass rhs) =>
+        !lhs.Equals(rhs);
+
+    public static bool operator >(Mass lhs, Mass rhs) =>
+        lhs.Kilograms > rhs.Kilograms;
+
+    public static bool operator <(Mass lhs, Mass rhs) =>
+        lhs.Kilograms < rhs.Kilograms;
+
+    public static bool operator >=(Mass lhs, Mass rhs) =>
+        lhs.Kilograms >= rhs.Kilograms;
+
+    public static bool operator <=(Mass lhs, Mass rhs) =>
+        lhs.Kilograms <= rhs.Kilograms;
+
+    public Mass Round() =>
+        new Mass(Math.Round(Kilograms));
+
+    public Mass Sqrt() =>
+        new Mass(Math.Sqrt(Kilograms));
+
+    public Mass Abs() =>
+        new Mass(Math.Abs(Kilograms));
+
+    public Mass Min(Mass rhs) =>
+        new Mass(Math.Min(Kilograms, rhs.Kilograms));
+
+    public Mass Max(Mass rhs) =>
+        new Mass(Math.Max(Kilograms, rhs.Kilograms));
+
+    #endregion
+
+    #region Conversion units
+
+    public double Grams => Value / 1000;
+    public double Kilograms => Value;
+    public double Tonnes => Value * 1000;
```

Comment:
Likewise, since `Value` is the number of Kilograms, the number of metric tonnes should be `Value / 1000` instead? (e.g. there are 1000 kilograms in a metric tonne).

Possibly should call this `MetricTonnes`?